modules/services/mpd-mpris.nix

```diff
@@ -0,0 +1,88 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.mpd-mpris;
+
+  mpdCfg = config.services.mpd;
+
+  renderArg = name: value:
+    if lib.isBool value && value then
+      "-${name}"
+    else if lib.isInt value then
+      "-${name} ${toString value}"
+    else if lib.isString value then
+      "-${name} ${lib.escapeShellArg value}"
+    else
+      "";
+
+  concatArgs = strings:
+    lib.concatStringsSep " " (lib.filter (s: s != "") strings);
+
+  renderArgs = args: concatArgs (lib.mapAttrsToList renderArg args);
+
+  renderCmd = pkg: args: "${pkg}/bin/mpd-mpris ${renderArgs args}";
+in {
+  meta.maintainers = [ lib.hm.maintainers.olmokramer ];
+
+  options.services.mpd-mpris = {
+    enable = lib.mkEnableOption
+      "mpd-mpris: An implementation of the MPRIS protocol for MPD";
+
+    package = lib.mkPackageOption pkgs "mpd-mpris" { };
+
+    mpd = {
+      host = lib.mkOption {
+        type = with lib.types; nullOr str;
+        default = if mpdCfg.enable then mpdCfg.network.listenAddress else null;
+        defaultText = "config.services.mpd.network.listenAddress";
+        example = "192.168.1.1";
+        description = "The address where MPD is listening for connections.";
+      };
+
+      port = lib.mkOption {
+        type = with lib.types; nullOr port;
+        default = if mpdCfg.enable then mpdCfg.network.port else null;
+        defaultText = "config.services.mpd.network.port";
+        description = ''
+          The port number where MPD is listening for connections.
+        '';
+      };
+
+      password = lib.mkOption {
+        type = with lib.types; nullOr str;
+        default = null;
+        description = ''
+          The password to connect to MPD.
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      (lib.hm.assertions.assertPlatform "services.mpd-mpris" pkgs
+        lib.platforms.linux)
+    ];
+
+    systemd.user.services.mpd-mpris = {
+      Install = { WantedBy = [ "default.target" ]; };
+
+      Unit = {
+        Description =
+          "mpd-mpris: An implementation of the MPRIS protocol for MPD";
+        After = [ "mpd.service" ];
```

Comment:
```suggestion
        After = [ "mpd.service" ];
        Requires = [ "mpd.service" ];
```

Reply:
I left the `Requires` out because MPD might run on another host. Maybe it should only be added if the `services.mpd.enable = true` and `services.mpd-mpris.mpd.{host,port}` are both `null`?

Reply:
I've added a `services.mpd-mpris.mpd.useLocal` option that defaults to `config.services.mpd.enable` and, when `true`, adds the `Required` field.